tests/hawc/apps/lit/test_api.py

```diff
@@ -661,3 +662,124 @@ def test_valid_requests(self, db_keys):
+        ref.refresh_from_db()
+
+        assert response.status_code == 403
+        # no user tag was created
+        assert ref.user_tags.count() == 2
+        assert ref.user_tags.filter(user__email="reviewer@hawcproject.org").exists() is False
+
+        user_tag_id = ref.user_tags.first().pk
+        response = client.post(resolve_conflict_url, data={"user_tag_id": user_tag_id})
+        ref.refresh_from_db()
+
+        assert response.status_code == 403
+        # tags were not added to the reference/conflict was not resolved
+        assert ref.tags.count() == 0
+        assert ref.has_user_tag_conflicts() is True
+
+    def test_valid_requests(self, db_keys):
+        ref: models.Reference = models.Reference.objects.get(id=db_keys.reference_untagged)
+        pm = HAWCUser.objects.get(email="pm@hawcproject.org")
+        tm = HAWCUser.objects.get(email="team@hawcproject.org")
+
+        pm_tags = [32]
+        tm_tags = [33]
+
+        update_tags_url = reverse("lit:api:reference-tag", args=(ref.pk,))
+        resolve_conflict_url = reverse("lit:api:reference-resolve-conflict", args=(ref.pk,))
+
+        client = APIClient()
+
+        assert ref.user_tags.count() == 0
+        assert ref.has_user_tag_conflicts() is False
+
+        # create a conflict by applying different tags as two different users
+        assert client.login(email=tm.email, password="pw") is True
+        response = client.post(update_tags_url, data={"tags": tm_tags}, format="json")
+        assert response.status_code == 200
+        client.logout()
+
+        assert client.login(email=pm.email, password="pw") is True
+        response = client.post(update_tags_url, data={"tags": pm_tags}, format="json")
+        assert response.status_code == 200
+
+        ref.refresh_from_db()
+        assert ref.user_tags.count() == 2
+        assert ref.has_user_tag_conflicts() is True
+        assert ref.tags.count() == 0
+
+        # resolve the conflict using Project Manager's tags
+        pm_user_tag = ref.user_tags.get(user=pm)
+        response = client.post(resolve_conflict_url, data={"user_tag_id": pm_user_tag.pk})
+        assert response.status_code == 200
+
+        ref.refresh_from_db()
+        assert ref.has_user_tag_conflicts() is False
+        assert list(ref.tags.values_list("id", flat=True)) == pm_tags
+
+        # test applying tags w/o conflict resolution
+        ref = models.Reference.objects.get(id=db_keys.reference_linked)
+        update_tags_url = reverse("lit:api:reference-tag", args=(ref.pk,))
+        tags = [2, 3]
+        response = client.post(update_tags_url, data={"tags": tags}, format="json")
+        assert response.status_code == 200
+
+        # tags are applied to the reference without creating a conflict
+        ref.refresh_from_db()
+        assert list(ref.tags.values_list("id", flat=True)) == tags
+        assert ref.has_user_tag_conflicts() is False
+
+    def test_bad_requests(self, db_keys):
```

Comment:
I split this up into two methods, one to test to tag api, and one for the conflict api. I did the same for above too